common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +466,131 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
     return true;
 }
 
+bool mon_wlan_hal_dwpal::dcs_scan_trigger(int dwell_time_msec,
+                                          const std::vector<unsigned int> &channel_pool)
+{
+    LOG(DEBUG) << __func__ << " received on interface=" << m_radio_info.iface_name;
+
+    //build scan parameters
+    ScanParams dcs_scan_params;
+    memset((void *)&dcs_scan_params, 0, sizeof(dcs_scan_params)); //must to ensure freq are zero
```

Comment:
Don't use `memset`. I think it is banned c function. Use the appropriate safec function, or simply use `ScanParams dcs_scan_params = {};`

Reply:
It must have been a remainder from when the code was C and not C++
Changed it to = { 0 } since we still need to zero out all the values before usage